br/pkg/stream/rewrite_meta_rawkv.go

```diff
@@ -787,18 +789,19 @@ func (sr *SchemasReplace) deleteRange(job *model.Job) error {
 			return errors.Trace(err)
 		}
 		if len(physicalTableIDs) > 0 {
-			// delete partition id instead of table id
-			for i := 0; i < len(physicalTableIDs); i++ {
-				newPid, exist := tableReplace.PartitionMap[physicalTableIDs[i]]
+			newPhysicalTableIDs := make([]int64, 0, len(physicalTableIDs))
+			// delete partition id
+			for _, oldPid := range physicalTableIDs {
+				newPid, exist := tableReplace.PartitionMap[oldPid]
 				if !exist {
-					log.Debug("DropTable/TruncateTable: try to drop a non-existent table, missing oldPartitionID",
-						zap.Int64("oldPartitionID", physicalTableIDs[i]))
+					logutil.CL(lctx).Warn("DropTable/TruncateTable: try to drop a non-existent table, missing oldPartitionID",
+						zap.Int64("oldPartitionID", oldPid))
 					continue
 				}
-				physicalTableIDs[i] = newPid
+				newPhysicalTableIDs = append(newPhysicalTableIDs, newPid)
 			}
-			if len(physicalTableIDs) > 0 {
-				sr.insertDeleteRangeForTable(newJobID, physicalTableIDs)
+			if len(newPhysicalTableIDs) > 0 {
+				sr.insertDeleteRangeForTable(newJobID, newPhysicalTableIDs)
```

Comment:
maybe add some info logs inside this function to record which id should be deleted.